src/shared/orgUtils.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2024, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { Connection } from '@salesforce/core';
+
+export class OrgUtils {
+  /**
+   * Given an app name, it queries the org to find the matching app id. To do so,
+   * it will first attempt at finding the app with a matching DeveloperName. If
+   * no match is found, it will then attempt at finding the app with a matching
+   * Label. If multiple matches are found, then the first match is returned.
+   *
```

Comment:
See [this discussion](https://salesforce.quip.com/lw7LAqDyuhjn#TUKADAFeHPe) which lead to the implementation here.